http/handler.go

```diff
@@ -389,25 +304,21 @@ func wrapGenericHandler(core *vault.Core, h http.Handler, props *vault.HandlerPr
 	hostname, _ := os.Hostname()
 
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+
+		nw := logical.NewStatusHeaderResponseWriter(w)
 		// This block needs to be here so that upon sending SIGHUP, custom response
 		// headers are also reloaded into the handlers.
 		if props.ListenerConfig != nil {
 			la := props.ListenerConfig.Address
 			listenerCustomHeaders := core.GetListenerCustomResponseHeaders(la)
 			if listenerCustomHeaders != nil {
-				w = &statusHeaderResponseWriter{
-					wrapped:     w,
-					logger:      core.Logger(),
-					wroteHeader: false,
-					statusCode:  200,
-					headers:     listenerCustomHeaders.StatusCodeHeaderMap,
-				}
+				nw.SetHeaders(listenerCustomHeaders.StatusCodeHeaderMap)
```

Comment:
Why introduce a method for this instead of just making it an argument to the constructor?

Reply:
Sorry, I'm not following.  What would break if we got rid of SetHeaders and instead assigned listenerCustomHeaders.StatusCodeHeaderMap to a local var here?  Then passed that var as a 2nd arg to logical.NewStatusHeaderResponseWriter?

My primary goal here is to minimize mutable state in the type.